finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -105,6 +105,14 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
         log.debug(e, "[%s] dispatcher closed", prefix)
         Future.exception(e)
 
+      case Throw(e: ChannelClosedException)
+      // if all streams have already been closed, then this just means that
+      // the client failed to send a GOAWAY frame...
+      if closed.get || streams.keys.asScala.forall { _ <= closedId.get } =>
```

Comment:
`closedId` is the highest closed stream id.  If we encounter a stream id that's not in our map, it might not be there because the stream was closed and removed from the map, or it might not be there because the stream hasn't been created yet.  closedId helps us differentiate those cases.  I think.